File: xchanger/src/main/java/com/finsim/xchanger/orders/model/Order.java
package com.finsim.xchanger.orders.model;

import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.finsim.xchanger.common.model.Price;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "orders")
public class Order {
    @Id
    private String id;

    private String dossier;
    private String isin;
    private String type;

    private Price price;
    private int quantity;
    private String options;

    private int leftQuantity;
    private List<OrderTransaction> orderTransactions;

    public OrderDto toDto() {
        return new OrderDto(
            this.id,
            this.dossier,
            this.isin,
            this.type,
            this.price,
            this.quantity,
            this.options,
            this.leftQuantity,
            this.orderTransactions
        );
    }
}

File: mainframe-lib/xchanger/model/order.go
package model

type Order struct {
	Id           string             `json:"id"`
	Dossier      string             `json:"dossier"`
	Isin         string             `json:"isin"`
	Type         string             `json:"type"`
	Price        Price              `json:"price"`
	Quantity     int32              `json:"quantity"`
	Options      string             `json:"options"`
	LeftQuantity int32              `json:"leftQuantity"`
	Transactions []OrderTransaction `json:"transactions,omitempty"`
}

type OrderTransaction struct {
	Dossier   string `json:"dossier"`
	Quantity  int32  `json:"quantity"`
	Price     Price  `json:"price"`
	Timestamp string `json:"timestamp"`
}

type InsertOrderInput struct {
	Dossier  string `json:"dossier"`
	Isin     string `json:"isin"`
	Type     string `json:"type"`
	Price    Price  `json:"price"`
	Quantity int32  `json:"quantity"`
	Options  string `json:"options"`
}
